### PR TITLE
🐛 Retrieve branch on release triggers

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/slsa-framework/slsa-verifier/issues/88

Note: I've added the e2e tests. They are failing (e.g., https://github.com/slsa-framework/slsa-github-generator/issues/208) until this PR is merged.

### CODE REVIEW

File: pkg/provenance.go [hunk 1]
```diff
@@ -642,17 +656,57 @@ func getBaseRef(environment map[string]interface{}) (string, error) {
...
-	eventPayload, ok := environment["github_event_payload"]
+	payload, err := getEventPayload(environment)
+	if err != nil {
+		return "", err
+	}
+
+	return getAsString(payload, "base_ref")
+}
+
+func getTargetCommittish(environment map[string]interface{}) (string, error) {
```
Comment (ianlewis) [inline pkg/provenance.go:667]: TIL https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish

File: pkg/provenance.go [hunk 2]
```diff
@@ -642,17 +656,57 @@ func getBaseRef(environment map[string]interface{}) (string, error) {
...
+	release, ok := releasePayload.(map[string]interface{})
 	if !ok {
-		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "parameters type payload")
+		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "parameters type releasePayload")
 	}
 
-	return getAsString(payload, "base_ref")
+	branch, err := getAsString(release, "target_commitish")
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", err, "target_commitish not present")
```
Comment (ianlewis) [inline pkg/provenance.go:695]: target_committish

**Review by ianlewis — APPROVED** [review]

File: pkg/provenance.go (hunk 2 above)
Comment (laurentsimon) [reply pkg/provenance.go:695]: the payload has a `target_commitish` field, but not `target_committish`. Not sure if it's intentional or a typo.

Comment (ianlewis) [reply pkg/provenance.go:695]: Ah, I see. It's actually a typo in the payload. Ok.

**Pull request merged**